drawBot/context/baseContext.py

```diff
@@ -886,6 +875,58 @@ def copy(self):
         return new
 
 
+def makeTextBoxes(attributedString, xy, align):
+    extraPadding = 20
+    x, y = xy
+    w, h = attributedString.size()
+    w += extraPadding
+
+    if align is not None:
+        attributedString = attributedString.mutableCopy()
+        # overwrite all align settings in each paragraph style
+        def block(value, rng, stop):
+            value = value.mutableCopy()
```

Comment:
Why this copy?

Reply:
this should not modify the original attributed string (which is also the source of the formattedString) 
The copy on the paragraphStyle is also necessary as this seems to not good to modify existing objects inside a attributed string and the copy on the attributed string seems not to be a deep copy where internal objects are also copied.